File: M5_GoLang_Assignments/Assignment_set_2/Blog_Management_System/config/dbconfig.go
package config

import (
	"database/sql"
	"log"

	_ "modernc.org/sqlite"
)

var DB *sql.DB

func InitializeDatabase() {
	var err error
	DB, err = sql.Open("sqlite", "./myblogs.db")
	if err != nil {
		log.Fatal(err)
	}

	// create blogs table
	_, err = DB.Exec(`CREATE TABLE IF NOT EXISTS blogs(
		id INTEGER PRIMARY KEY AUTOINCREMENT,
		title TEXT,
		content TEXT,
		author TEXT
		);`)

	if err != nil {
		log.Fatal(err)
	}
}

func GetDB() *sql.DB {
	return DB
}

File: M3_Python_Assignments/Python_RestAPI/dbconfig.py
import sqlite3

def init_db():
    # Connect to SQLite database (or create it if it doesn't exist)
    conn = sqlite3.connect('bookbuddy.db')
    cursor = conn.cursor()
    
    # Create the 'books' table
    cursor.execute('''
    CREATE TABLE IF NOT EXISTS books (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        title TEXT NOT NULL,
        author TEXT NOT NULL,
        published_year INTEGER NOT NULL,
        genre TEXT NOT NULL
    );
    ''')
    
    # Insert sample data
    cursor.executemany('''
    INSERT INTO books (title, author, published_year, genre) 
    VALUES (?, ?, ?, ?);
    ''', [
        ('The Great Gatsby', 'F. Scott Fitzgerald', 1925, 'Fiction'),
        ('To Kill a Mockingbird', 'Harper Lee', 1960, 'Fiction'),
        ('1984', 'George Orwell', 1949, 'Sci-Fi')
    ])
    
    # Commit and close the connection
    conn.commit()
    conn.close()

# Run the function to initialize the database
if __name__ == "__main__":
    init_db()
    print("Database and sample data initialized.")
